apps/glusterfs/app_device.go

```diff
@@ -220,20 +220,23 @@ func (a *App) DeviceDelete(w http.ResponseWriter, r *http.Request) {
 			return logger.Err(err)
 		}
 
-		// Check if we can delete the device
-		if device.HasBricks() {
-			http.Error(w, device.ConflictString(), http.StatusConflict)
-			logger.LogError(device.ConflictString())
-			return ErrConflict
-		}
-
 		// Access node entry
 		node, err = NewNodeEntryFromId(tx, device.NodeId)
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return logger.Err(err)
 		}
 
+		// Check if we can delete the device
+		if err := device.CheckDelete(tx); err != nil {
+			if err.Error() != ErrConflict.Error() {
```

Comment:
I'm not sure why you are comparing the error strings. Could you elaborate if there's a specific reason for this?

Reply:
@phlogistonjohn am comparing the error to return the HTTP status code.
if the device state is not in failed state am considering it as a bad request

Reply:
I mean why is the line `if err.Error() != ErrConflict.Error() {` and not `if err != ErrConflict` ? ErrConflict is a error constant after all.

Reply:
yeah i could have just compared err instead of comparing string. updated PTAL